uni-stark/src/prover.rs

```diff
@@ -83,32 +129,134 @@ where
                 pcs.coset_shift().exp_power_of_2(log_quotient_degree),
             )
         });
+    (quotient_commit, quotient_data)
+}
+
+pub fn get_trace_lde<SC>(config: &SC, trace: RowMajorMatrix<SC::Val>) -> Commitment<SC>
+where
+    SC: StarkConfig,
+{
+    let pcs = config.pcs();
+    let (trace_commit, trace_data) =
+        info_span!("commit to trace data").in_scope(|| pcs.commit_batch(trace));
+
+    (trace_commit, trace_data)
+}
+type LogQuotientDegree = usize;
+type LogDegree = usize;
+pub fn get_trace_and_quotient_ldes<SC, A>(
+    config: &SC,
+    trace: RowMajorMatrix<SC::Val>,
+    air: &A,
+    challenger: &mut SC::Challenger,    
+) -> (ProverData<SC>, ProverData<SC>, LogQuotientDegree, LogDegree)
+where
+    SC: StarkConfig,
+    A: Air<SymbolicAirBuilder<SC::Val>> + for<'a> Air<ProverConstraintFolder<'a, SC>>,
+{
+    let log_degree = log2_strict_usize(trace.height());
+
+    // The quotient's actual degree is approximately (max_constraint_degree - 1) n,
+    // where subtracting 1 comes from division by the zerofier.
+    // But we pad it to a power of two so that we can efficiently decompose the quotient.
+    let log_quotient_degree = log2_ceil_usize(get_max_constraint_degree(air) - 1);
+
+    let (trace_commit, trace_data) = get_trace_lde::<SC>(config, trace);
+
+    challenger.observe(trace_commit.clone());
+
+    let (quotient_commit, quotient_data) = get_quotient::<SC, A>(
+        log_degree,
+        log_quotient_degree,
+        config,
+        air,
+        challenger,
+        &trace_data,
+    );
+
     challenger.observe(quotient_commit.clone());
 
+    (trace_data, quotient_data, log_degree, log_quotient_degree)
+}
+
+pub fn get_trace_and_quotient_commitments<SC, A>(
+    config: &SC,
+    trace: RowMajorMatrix<SC::Val>,
+    log_quotient_degree: usize,
+    challenger: &mut SC::Challenger,
+    air: &A,
+) -> (Commitment<SC>, Commitment<SC>)
+where
+    SC: StarkConfig,
+    A: Air<SymbolicAirBuilder<SC::Val>> + for<'a> Air<ProverConstraintFolder<'a, SC>>,
+{
+    let log_degree = log2_strict_usize(trace.height());
+
+    // The quotient's actual degree is approximately (max_constraint_degree - 1) n,
+    // where subtracting 1 comes from division by the zerofier.
+    // But we pad it to a power of two so that we can efficiently decompose the quotient.
+
+    let (trace_commit, trace_data) = get_trace_lde::<SC>(config, trace);
+
+    challenger.observe(trace_commit.clone());
+
+    let (quotient_commit, quotient_data) = get_quotient::<SC, A>(
+        log_degree,
+        log_quotient_degree,
+        config,
+        air,
+        challenger,
+        &trace_data,
+    );
+
+    challenger.observe(quotient_commit.clone());
+
+    ((trace_commit, trace_data), (quotient_commit, quotient_data))
+}
+
+    #[instrument(skip_all)]
```

Comment:
Looks like an indentation error here and line 219?

Reply:
I think this was fixed by latest commit